blocks_vertical/control.js

```diff
@@ -183,57 +183,41 @@ Blockly.Blocks['control_if_else'] = {
   }
 };
 
-Blockly.Blocks['control_stop_menu'] = {
-  /**
-   * Stop drop-down menu.
-   * @this Blockly.Block
-   */
-  init: function() {
-    this.jsonInit(
-      {
-        "message0": "%1",
-        "args0": [
-          {
-            "type": "field_dropdown",
-            "name": "STOP_OPTION",
-            "options": [
-              ['all', 'all'],
-              ['this script', 'this script'],
-              ['other scripts in sprite', 'other scripts in sprite']
-            ]
-          }
-        ],
-        "inputsInline": true,
-        "output": "String",
-        "outputShape": Blockly.OUTPUT_SHAPE_ROUND,
-        "colour": Blockly.Colours.control.secondary,
-        "colourSecondary": Blockly.Colours.control.secondary,
-        "colourTertiary": Blockly.Colours.control.tertiary
-      });
-  }
-};
-
 Blockly.Blocks['control_stop'] = {
   /**
    * Block for stop all scripts.
    * @this Blockly.Block
    */
   init: function() {
-    this.jsonInit({
-      "id": "control_stop",
-      "message0": "stop %1",
-      "args0": [
-        {
-          "type": "input_value",
-          "name": "STOP_OPTION"
-        }
-      ],
-      "inputsInline": true,
-      "previousStatement": null,
-      "colour": Blockly.Colours.control.primary,
-      "colourSecondary": Blockly.Colours.control.secondary,
-      "colourTertiary": Blockly.Colours.control.tertiary
+    var stopDropdown = new Blockly.FieldDropdown([
+        ['all', 'all'],
+        ['this script', 'this script'],
+        ['other scripts in sprite', 'other scripts in sprite']
+    ], function(option) {
+      if (option == 'other scripts in sprite') {
```

Comment:
this.sourceBlock_.setNextStatement(option == 'other scripts in sprite');

And at least the keys should be constants.